rust/headless-client/src/lib.rs

```diff
@@ -50,7 +50,7 @@ struct Cli {
     // TODO: It isn't good for security to pass the token as a CLI arg.
     // If we pass it as an env var, we should remove it immediately so that
     // other processes don't see it. Reading it from a file is probably safest.
-    #[arg(env = "FIREZONE_TOKEN")]
+    #[arg(env = "FIREZONE_TOKEN", hide = true)]
```

Comment:
Wondering if there's any usability downside to this. I think env vars are standard even for sensitive things like [api keys](https://tailscale.com/kb/1210/terraform-provider).

Reply:
I fixed the comment. I think you're right that normally a non-root user shouldn't be able to see env vars of a root process.

And this PR doesn't disable the CLI arg or env var, it just prevents clap from printing it when you run `--help`

I get why they do that, but it feels like the wrong default for secrets.

<img width="628" alt="image" src="https://github.com/firezone/firezone/assets/13400041/314e4885-1ed0-4c42-9340-7b7b71e82cd0">